src/shared/Microsoft.Git.CredentialManager.Tests/Interop/LibGit2Tests.cs

```diff
@@ -0,0 +1,201 @@
+            string fileL1Path = Path.Combine(directoryL0Path, "inner-file.txt");
+            string directoryL1Path = Path.Combine(directoryL0Path, "sub-directory");
+
+            var git = new CredentialManager.Interop.LibGit2();
+
+            // Create files and directories
+            Directory.CreateDirectory(directoryL0Path);
+            Directory.CreateDirectory(directoryL1Path);
+            File.WriteAllText(fileL0Path, string.Empty);
+            File.WriteAllText(fileL1Path, string.Empty);
+
+            // Check from L0 file
+            string fileL0RepoPath = git.GetRepositoryPath(fileL0Path);
+            AssertPathsEquivalent(expectedRepoGitPath, fileL0RepoPath);
+
+            // Check from L0 directory
+            string dirL0RepoPath = git.GetRepositoryPath(directoryL0Path);
+            AssertPathsEquivalent(expectedRepoGitPath, dirL0RepoPath);
+
+            // Check from L1 file
+            string fileL1RepoPath = git.GetRepositoryPath(fileL1Path);
+            AssertPathsEquivalent(expectedRepoGitPath, fileL1RepoPath);
+
+            // Check from L1 directory
+            string dirL1RepoPath = git.GetRepositoryPath(directoryL1Path);
+            AssertPathsEquivalent(expectedRepoGitPath, dirL1RepoPath);
+        }
+
+        [Fact]
+        public void LibGit2Configuration_GetString_DoesNotExists_ReturnsNull()
+        {
+            string repoPath = CreateRepository();
+
+            var git = new CredentialManager.Interop.LibGit2();
+            using (var config = git.GetConfiguration(repoPath))
+            {
+                string randomKey = $"{Guid.NewGuid():N}.{Guid.NewGuid():N}";
+                string value = config.GetString(randomKey);
+                Assert.Null(value);
+            }
+        }
+
+        #region Test helpers
+
+        private static void AssertPathsEquivalent(string expected, string actual)
+        {
+            string realExpected = RealPath(expected);
+            string realActual = RealPath(actual);
+
+            Assert.Equal(realExpected, realActual);
+        }
+
+        private static string RealPath(string path)
+        {
+            if (PlatformUtils.IsPosix())
+            {
+                byte[] pathBytes = Encoding.UTF8.GetBytes(path);
+                byte[] resolvedBytes = new byte[64 * 1024];
+                unsafe
+                {
+                    fixed (byte* pathPtr = pathBytes)
+                    fixed (byte* resolvedPtr = resolvedBytes)
+                    {
+                        if (Stdlib.realpath(pathPtr, resolvedPtr) == IntPtr.Zero)
+                        {
+                            return null;
+                        }
+                    }
+                }
+
+                int nullByteIndex = Array.IndexOf(resolvedBytes, (byte) '\0');
+                return Encoding.UTF8.GetString(resolvedBytes, 0, nullByteIndex);
+            }
+
+            if (PlatformUtils.IsWindows())
```

Comment:
nit: `else if`?

Reply:
Not required since all code paths from the above `if` block `return`. I actually get a warning that the `else` here is redundant.